api/src/opentrons/hardware_control/modules/update.py

```
@@ -171,9 +172,9 @@ def _format_avrdude_response(raw_response: str) -> Tuple[bool, str]:
             ports = await _discover_ports()
             if ports:
                 discovered_ports = list(filter(
-                    lambda x: x.endswith('bootloader'), ports))
+                    lambda x: 'bootloader' in x, ports))
                 if len(discovered_ports) == 1:
```

should we raise an error if this isn't 1?

---

That sounds like a good protection for now. I'd like to update this flow to be a little smarter about tracking the relationship between a device port and it's bootloader port. Though that will come in a follow up.